content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/components/common/cifproductfield/clientlibs/cifproductpicker.js

```diff
@@ -0,0 +1,192 @@
+
+        if (state.el) {
+            if (!state.open) {
+                show(control, state, handleSelections);
+            }
+        } else {
+            if (!pickerSrc) {
+                return;
+            }
+
+            state.loading = true;
+            Granite.$.ajax({
+                url: pickerSrc,
+                cache: false
+            })
+                .then(function(html) {
+                    return Granite.$(window)
+                        .adaptTo('foundation-util-htmlparser')
+                        .parse(html);
+                })
+                .then(function(fragment) {
+                    return Granite.$(fragment).children()[0];
+                })
+                .then(
+                    function(pickerEl) {
+                        state.el = pickerEl;
+                        state.api = Granite.$(pickerEl).adaptTo('foundation-picker');
+
+                        show(control, state, handleSelections);
+                    },
+                    function() {
+                        state.loading = false;
+                    }
+                );
+        }
+    };
+
+    var clickActivator = function(e) {
+        e.preventDefault();
+
+        var control = Granite.$(document).find(relActivator);
+        var pickerConfig = control.data();
+        var pickerSrc = pickerConfig.pickersrc;
+        if (!pickerSrc) {
+            var root = pickerConfig.root;
+            if (!root) {
+                root = defaultRootPath;
+            }
+            var filter = pickerConfig.filter;
+            if (!filter) {
+                filter = defaultFilter;
+            }
+            var selectionCount = pickerConfig.selectioncount;
+            if (!selectionCount) {
+                selectionCount = defaultSelectionCount;
+            }
+            var selectionId = pickerConfig.selectionid;
+            if (!selectionId) {
+                selectionId = defaultSelectionId;
+            }
+
+            pickerSrc =
+                defaultPickerSrc +
+                '?root=' +
+                root +
+                '&filter=' +
+                filter +
+                '&selectionCount=' +
+                selectionCount +
+                '&selectionId=' +
+                selectionId;
+        }
+
+        var handleSelections = function(selections) {
+            control.trigger(eventType, { selections: selections });
+        };
+
+        cifProductPicker(control, pickerSrc, handleSelections);
+    };
+
+    Granite.$(document).on('click', relActivator, clickActivator);
+
+    window.CIF.CifProductPicker.clickActivator = clickActivator;
+    window.CIF.CifProductPicker.getState = getState;
+    window.CIF.CifProductPicker.relActivator = relActivator;
+    window.CIF.CifProductPicker.cifProductPicker = cifProductPicker;
+    window.CIF.CifProductPicker.close = close;
+    window.CIF.CifProductPicker.show = show;
+})(window, document, window.CIF.Granite ? window.CIF.Granite : Granite);
```

Comment:
Is `window.CIF.Granite` a mock library? If so, it would introduce some confusion since the `window.CIF` namespace is used by the actual client-side components.

Again, the solution is not to pass the testing objects to the actual library, but expose the functions so that you can test them outside of the context.

Reply:
@dplaton , I think I have fixed this now.